astropy/coordinates/tests/test_representation.py

```diff
@@ -550,6 +550,23 @@ def test_init_array_nocopy(self):
         assert_allclose_quantity(y, s1.y)
         assert_allclose_quantity(z, s1.z)
 
+    def test_xyz_is_view_if_possible(self):
+        xyz = np.arange(1., 10.).reshape(3, 3)
+        s1 = CartesianRepresentation(xyz, unit=u.kpc, copy=False)
+        s1_xyz = s1.xyz
+        assert s1_xyz.value[0, 0] == 1.
+        xyz[0, 0] = 0.
+        assert s1.x[0] == 0.
+        assert s1_xyz.value[0, 0] == 0.
+        # Not possible: we don't check that tuples are from the same array
+        xyz = np.arange(1., 10.).reshape(3, 3)
+        s2 = CartesianRepresentation(*xyz, unit=u.kpc, copy=False)
```

Comment:
Note that in principle, it would be possible to detect this, since one would have `x.base is y.base and x.base is z.base and x.base[0].__array_interface__ == x.__array_interface__ and [same for y,z]`.